website/versioned_docs/version-v1.x/contracts.md

```diff
@@ -229,68 +251,111 @@ emptyBallotRoots[3] = uint256(12031563002271722465187541954825013132282571927669
+This contract is used to prepare parameters for the zk-SNARK circuits as well as for verifying proofs. It should be deployed alongside `MACI` and ownership assigned to the coordinator.
+It will process messages in batches, and after all batches have been processed, the `sbCommitment` can then be used for the `Tally` and `Subsidy` contracts.
 
 ## Tally
 
-The Tally contract is used during votes tallying and by users to verify the tally results.
+::info
+The `Tally` contract is present also in a non quadratic voting fashion, and is slightly smaller due to not having the `verifyPerVOSpentVoiceCredits` function. This is not required because with normal (non quadratic) voting, each vote by a user is not the square root of the voice credits spent.  
+:::
+
+The `Tally` contract is used by the coordinator to submit commitments to the tally results via the `tallyVotes` function. This is done in batches and the final commitment can be used by the users to verify the validity of the results.
+
+Below are the functions which users can use to verify the results:
 
-## Subsidy
+- `verifySpentVoiceCredits`
+- `verifyPerVOSpentVoiceCredits`
+- `verifyTallyResult`
 
-This contract is used to verify that the subsidy calculations are correct. It is also used to update the subsidy commitment if the proof is valid.
+## Subsidy (optional)
 
-## SignUpToken
+The subsidy contract can be used by the coordinator to post proofs of valid subsidy calcuations, as well as users to verify that the subsidy calculations were performed correctly.
 
-This contract should be used by the SignUpGateKeeper to determine whether a user is allowed to register. The default contract provided with MACI is a simple ERC721 token. Coordinators can use this contract to mint a token for each of the participants in the voting process.
+## SignUpToken (optional)
+
+This contract can be used by the `SignUpGateKeeper` to determine whether a user is allowed to register. The default contract provided with MACI is a simple ERC721 token. Coordinators can use this contract to mint a token for each of the participants in the voting process, and gatekeep access to a round by using a `SignUpTokenGatekeeper` contract.
 
 ## SignUpGatekeeper
 
-MACI requires a signup gatekeeper to ensure that only designed users register. It is up to MACI's deployer how they wish to allow sign-ups, therefore they can implement their own GateKeeper. The repository comes with two presets:
+MACI requires a signup gatekeeper to ensure that only designed users register. It is up to MACI's deployer how they wish to allow sign-ups, therefore they can implement their own GateKeeper. The repository comes with different options:
 
 - `FreeForAllSignUpGatekeeper` - This allows anyone to signup on MACI.
 - `SignUpTokenGatekeeper` - This makes use of a ERC721 token to gatekeep the signup function.
+- `EASGatekeeper` - This allows to gatekeep signups to only users which have a specific EAS attestation.
```

Comment:
might be easier to read --> This allows gatekeeping signups to only users who have a specific EAS attestation.